File: java_app/src/main/java/com/petnagy/java/example/injection/DaggerActivityModule.java
package com.petnagy.java.example.injection;

import com.petnagy.java.example.MainActivity;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public interface DaggerActivityModule {

    @PerActivity
    @ContributesAndroidInjector
    MainActivity contributeMaineActivity();

}

File: app/src/main/java/com/petnagy/app/injection/DaggerActivityModule.kt
package com.petnagy.app.injection

import com.petnagy.app.MainActivity
import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
interface DaggerActivityModule {

    @PerActivity
    @ContributesAndroidInjector
    fun contributeMainActivity(): MainActivity

}